openupgradelib/openupgrade.py

```diff
@@ -513,10 +513,13 @@ def rename_columns(cr, column_spec):
                         table, old, new)
             cr.execute(
                 'ALTER TABLE "%s" RENAME "%s" TO "%s"' % (table, old, new,))
-            cr.execute(
-                'ALTER INDEX IF EXISTS "%s_%s_index" RENAME TO "%s_%s_index"' %
-                (table, old, table, new)
-            )
+            old_index_name = "%s_%s_index" % (table, old)
+            new_index_name = "%s_%s_index" % (table, new)
+            if len(new_index_name) <= 63:
```

Comment:
Shouldn't this check as well in `len(old_index_name) <= 63`?

Reply:
A check wouldn't hurt, but since the index is there, if it were > 63 chars wouldn't it be already be truncated?

Reply:
Yes, and that's why the renaming would fail then, as it won't find the previous index name we are providing (as it's not truncated).

Reply:
What about something like this:

            max_len = 63 - len(new + '_index') # PostgreSQL's Max Identifier Length Is 63 Bytes (NAMEDATALEN - 1)
            renamed_table = (table[:max_len]) if len(table) > max_len else table
            cr.execute(
                'ALTER INDEX IF EXISTS "%s_%s_index" RENAME TO "%s_%s_index"' %
                (table, old, renamed_table, new)
            )

Reply:
> it won't find the previous index name we are providing (as it's not truncated).

Yes, it finds it because it is truncated before searching for it too. I tested that.

Reply:
Are you talking that is truncated in Postgres although you pass a name over 63 chars?

Reply:
Yes

Reply:
Then I don't get why we need to shorten new index name and not the old one...

Reply:
It's not shortening it; it's just testing if it is longer that what is supported and, if so, skipping the rename. The shortening is done by postgres automatically.